File: server/app/schemas/onboarding.py
"""Onboarding schemas."""

from pydantic import BaseModel, EmailStr

from app.schemas.base import CamelModel


class OnboardingProfileRequest(BaseModel):
    name: str
    avatar_url: str | None = None


class OnboardingTeamRequest(BaseModel):
    emails: list[EmailStr]


class CriterionInput(BaseModel):
    name: str
    description: str = ""
    weight: int = 20
    question_count: int = 3


class OnboardingCandidateInput(BaseModel):
    name: str
    email: str


class OnboardingRoleRequest(BaseModel):
    title: str
    department: str
    seniority: str
    location: str
    criteria: list[CriterionInput]
    candidate: OnboardingCandidateInput | None = None


class OnboardingCompleteResponse(CamelModel):
    message: str
    role_id: str
    interview_link: str | None = None

File: client/src/api/onboarding.ts
import client from "./client";

export interface ProfilePayload {
  name: string;
  avatar_url?: string;
}

export interface TeamPayload {
  emails: string[];
}

export interface CriterionInput {
  name: string;
  description: string;
  weight: number;
  question_count: number;
}

export interface CandidateInput {
  name: string;
  email: string;
}

export interface RolePayload {
  title: string;
  department: string;
  seniority: string;
  location: string;
  criteria: CriterionInput[];
  candidate?: CandidateInput;
}

export interface OnboardingCompleteResponse {
  message: string;
  roleId: string;
  interviewLink: string | null;
}

export const onboardingApi = {
  completeProfile: (payload: ProfilePayload) =>
    client.put("/api/onboarding/profile", payload),
  inviteTeam: (payload: TeamPayload) =>
    client.post("/api/onboarding/team", payload),
  createFirstRole: (payload: RolePayload) =>
    client.post<OnboardingCompleteResponse>("/api/onboarding/role", payload),
};
